controllers/components/codeflare/codeflare_controller.go

```diff
@@ -19,40 +19,64 @@ package codeflare
-	"sigs.k8s.io/controller-runtime/pkg/log"
 
 	componentsv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/components/v1"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/actions/deploy"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/actions/render/kustomize"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/actions/updatestatus"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/predicates/resources"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/reconciler"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/metadata/labels"
 )
 
 // CodeFlareReconciler reconciles a CodeFlare object.
-type CodeFlareReconciler struct {
-	client.Client
-	Scheme *runtime.Scheme
-}
 
-// Reconcile is part of the main kubernetes reconciliation loop which aims to
-// move the current state of the cluster closer to the desired state.
-// TODO(user): Modify the Reconcile function to compare the state specified by
-// the CodeFlare object against the actual cluster state, and then
-// perform operations to make the cluster state reflect the state specified by
-// the user.
-//
-// For more details, check Reconcile and its Result here:
-// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.12.2/pkg/reconcile
-func (r *CodeFlareReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
-
-	// TODO(user): your logic here
-
-	return ctrl.Result{}, nil
-}
+func (s *componentHandler) NewComponentReconciler(ctx context.Context, mgr ctrl.Manager) error {
+	_, err := reconciler.ComponentReconcilerFor(
+		mgr,
+		componentsv1.CodeFlareInstanceName,
+		&componentsv1.CodeFlare{},
+	).
+		// customized Owns() for Component with new predicates
+		Owns(&corev1.ConfigMap{}).
+		Owns(&corev1.Secret{}).
+		Owns(&rbacv1.ClusterRoleBinding{}).
+		Owns(&rbacv1.ClusterRole{}).
+		Owns(&rbacv1.Role{}).
+		Owns(&rbacv1.RoleBinding{}).
+		Owns(&corev1.ServiceAccount{}).
+		Owns(&corev1.Service{}).
+		Owns(&admissionregistrationv1.MutatingWebhookConfiguration{}).
+		Owns(&admissionregistrationv1.ValidatingWebhookConfiguration{}).
+		Owns(&appsv1.Deployment{}, reconciler.WithPredicates(resources.NewDeploymentPredicate())).
+		Watches(&extv1.CustomResourceDefinition{}).
+		// Add CodeFlare-specific actions
+		WithAction(initialize).
+		WithAction(devFlags).
+		WithAction(kustomize.NewAction(
+			kustomize.WithCache(),
+			kustomize.WithLabel(labels.ODH.Component(ComponentName), "true"),
+			kustomize.WithLabel(labels.K8SCommon.PartOf, ComponentName),
+		)).
+		WithAction(deploy.NewAction(
+			deploy.WithCache(),
+			deploy.WithFieldOwner(componentsv1.CodeFlareInstanceName),
+			deploy.WithLabel(labels.ComponentPartOf, componentsv1.CodeFlareInstanceName),
+		)).
+		WithAction(updatestatus.NewAction(
+			updatestatus.WithSelectorLabel(labels.ComponentPartOf, componentsv1.CodeFlareInstanceName),
+		)).
```

Comment:
we probabaly missed the gc part?

Reply:
@zdtsw Can you help explain a little bit more?

Reply:
see https://github.com/opendatahub-io/opendatahub-operator/blob/feature-operator-refactor/controllers/components/dashboard/dashboard_controller.go#L116